src/ansys/simai/core/data/model_configuration.py

```diff
@@ -330,6 +327,72 @@ def __get_build_preset(self):
 
     build_preset = property(__get_build_preset, __set_build_preset)
 
+    def __validate_variables(self, vars_to_validate: list[str], var_type: str):
+        sample_metadata = self.project.sample.fields.get("extracted_metadata")
+        var_fields = dict_get(sample_metadata, var_type, "fields", default=[])
+        var_fields_name = {fd.get("name") for fd in var_fields}
+        unknown_variables = set(vars_to_validate) - var_fields_name
+        if unknown_variables:
+            raise ProcessingError(
+                f"{self.__class__.__name__}: {var_type} variables {', '.join(unknown_variables)} do not exist in the reference sample."
+            ) from None
+
+    def __validate_surface_variables(self, vars_to_validate: list[str]):
+        sample_metadata = self.project.sample.fields.get("extracted_metadata")
+        if not sample_metadata.get("volume"):
+            raise ProcessingError(
+                "No volume file was found in the reference sample. A volume file is required to use volume variables."
+            ) from None
+        self.__validate_variables(vars_to_validate, "surface")
+
+    def __validate_volume_variables(self, vars_to_validate: list[str]):
+        sample_metadata = self.project.sample.fields.get("extracted_metadata")
+        if not sample_metadata.get("volume"):
+            raise ProcessingError(
+                "No volume file was found in the reference sample. A volume file is required to use volume variables."
+            ) from None
+        self.__validate_variables(vars_to_validate, "volume")
+
+    def __set_input(self, model_input: ModelInput):
+        if not model_input:
+            model_input = ModelInput()
```

Comment:
`model_input` is not marked as optional. Also with this if I do `model_config.input = None`, input is not gonna be `None` which is not what I would expect. Granted that `None` is not a valid value so maybe we should just raise instead ?

Reply:
`input` is set as optional in the constructor of `ModelConfiguration` see [here](https://github.com/ansys/pysimai/blob/9d80677d6e297bade43d4f30fe8dc7d35388b262/src/ansys/simai/core/data/model_configuration.py#L342), so it could get an empty `ModelInput` but not `None` indeed. I changed it so it raises an error instead